realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -142,6 +147,50 @@
     }
 
     /**
+     * Initializes the Realm library. It is required to call this method before interacting with any other of the Realm
+     * API's.
+     *
+     * A good place is in an {@link android.app.Application} subclass:
+     * <pre>
+     * {@code
+     * public class MyApplication extends Application {
+     *   \@Override
+     *   public void onCreate() {
+     *     super.onCreate();
+     *     Realm.init(this);
+     *   }
+     * }
+     * }
+     * </pre>
+     *
+     * Remember to register it in the {@code AndroidManifest.xml} file:
+     * <pre>
+     * {@code
+     * <?xml version="1.0" encoding="utf-8"?>
+     * <manifest xmlns:android="http://schemas.android.com/apk/res/android" package="io.realm.example">
+     * <application android:name=".MyApplication">
+     *   // ...
+     * </application>
+     * </manifest>
+     * }
+     * </pre>
+     *
+     * @param context the Application Context.
+     * @throws IllegalArgumentException if null context is provided.
+     */
+    public static synchronized void init(Context context) {
+        if (BaseRealm.applicationContext == null) {
+            if (context == null) {
+                throw new IllegalArgumentException("Non-null context required.");
+            }
+            BaseRealm.applicationContext = context.getApplicationContext();
```

Comment:
`BaseRealm.applicationContext = context.getApplicationContext();` is better to be called after `RealmCore.loadLibrary(BaseRealm.applicationContext);` since after it set, it is allowed to get a Realm instance in other threads, but `.so` might not be loaded yet.